src/widgets/analytics/analytics.ts

```diff
@@ -76,6 +77,13 @@ const analytics: AnalyticsWidget = function analytics(widgetParams) {
     throw new Error(withUsage('The `pushFunction` option is required.'));
   }
 
+  warning(
+    false,
+    `\`analytics\` widget has been deprecated. It is still supported in 4.x releases, but not further. It is replaced by \`insights\` middleware.
```

Comment:
we should show here exactly how to replace it IMO, the analytics widget gets fired on different moments than the insights one I think. Maybe we can look at how magento uses it, and migrate there to the insights middleware before deprecating? Maybe adding a FAQ or a guide in "upgrade guides"?

Reply:
As soon as the doc PR is merged & deployed, I will add the link here.